### PR TITLE
test(app): Android test for auth code flow

### PR DESCRIPTION
**Pull request by talwinder50**

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: demo/app/android/app/build.gradle [hunk 1]
```diff
@@ -92,4 +93,5 @@ dependencies {
     androidTestImplementation "androidx.test.ext:truth:1.5.0"
     androidTestImplementation "androidx.test:runner:1.2.0"
     androidTestUtil "androidx.test:orchestrator:1.4.2"
-}
+    implementation 'com.squareup.okhttp3:okhttp:4.9.0'
+}
```
Comment (birtony) [inline demo/app/android/app/build.gradle:97]: Missing extra empty line here?

File: demo/app/android/app/src/androidTest/kotlin/integration_test.kt [hunk 2]
```diff
@@ -106,4 +112,86 @@ class IntegrationTest {
...
+                        .build()
+                    val response2 = client.newCall(request2).clone().execute()
+                    location = response2.headers["location"]
+                    assertThat(location).contains("127.0.0.1")
+                    var issuedCreds = ciInteraction.requestCredentialWithAuth(userDID.assertionMethod(), location, null)
+                    assertThat(issuedCreds.length()).isGreaterThan(0)
+                }
+            }
+    }
+}
```
Comment (birtony) [inline demo/app/android/app/src/androidTest/kotlin/integration_test.kt:197]: And here?

File: demo/app/android/build.gradle [hunk 3]
```diff
@@ -78,3 +78,9 @@ subprojects {
 task clean(type: Delete) {
     delete rootProject.buildDir
 }
+tasks.withType(Test) {
+     testLogging {
+           showStandardStreams = true
+           events = ["passed", "failed", "skipped"]
+       }
+}
```
Comment (birtony) [inline demo/app/android/build.gradle:86]: Here as well

File: demo/app/android/gradle.properties [hunk 4]
```diff
@@ -3,3 +3,5 @@ android.useAndroidX=true
 android.enableJetifier=true
 INITIATE_ISSUANCE_URL=
 INITIATE_VERIFICATION_URL=
+INITIATE_ISSUANCE_URLS_AUTH_CODE_FLOW=
+ktor_version=2.3.2
```
Comment (birtony) [inline demo/app/android/gradle.properties:7]: And here

File: scripts/android_test.sh [hunk 5]
```diff
@@ -8,11 +8,12 @@
...
 echo "INITIATE_ISSUANCE_URL:${INITIATE_ISSUANCE_URL}"
 echo "INITIATE_VERIFICATION_URL:${INITIATE_VERIFICATION_URL}"
-
+echo "INITIATE_ISSUANCE_URLS_AUTH_CODE_FLOW:${INITIATE_ISSUANCE_URLS_AUTH_CODE_FLOW}"
 
 cd ../../demo/app/android
 adb reverse tcp:8075 tcp:8075 && adb reverse tcp:8072 tcp:8072 &&  adb reverse tcp:9229 tcp:9229 && \
-  ./gradlew app:connectedAndroidTest -PINITIATE_ISSUANCE_URL="${INITIATE_ISSUANCE_URL}" -PINITIATE_VERIFICATION_URL="${INITIATE_VERIFICATION_URL}"
\ No newline at end of file
+  ./gradlew app:connectedAndroidTest -PINITIATE_ISSUANCE_URL="${INITIATE_ISSUANCE_URL}" -PINITIATE_VERIFICATION_URL="${INITIATE_VERIFICATION_URL}" -PINITIATE_ISSUANCE_URLS_AUTH_CODE_FLOW="${INITIATE_ISSUANCE_URLS_AUTH_CODE_FLOW}"
```
Comment (birtony) [inline scripts/android_test.sh:19]: And here

**Review by birtony — APPROVED** [review]
Looks good, just pointed out some files are missing empty lines in the end

**Pull request merged**